### DISCUSSION TITLE
temporary .mm~ files not automatically deleted after closing FreePlane

### DISCUSSION THREAD

**Discussion by Olifair** [freeplane/freeplane#1645] [Help]
Hello everybody, 
I had posted this as a bug, but perhaps it is only a problem that can be resolved with a little help from the discussion group here :
When I close FreePlane, then the temporary .mm~ files are not deleted anmore automatically, so when I re-start FreePlane, it will complain the the files are already edited and will open them only in read-only mode (if I restart FreePlane during the same day), or it will complain that these files are old and thus the lock will be removed (when I restart FreePlane the next day). And it does this for every map in a separate window (which is totally normal). But as I have usually about 20 maps open in different tabs, this needs a lot of clicking before FreePlane is actually started. I then need to re-close, remove manually the .mm~ files and restart FreePlane. I have tried for some of the maps to change them and save them, but this does not change the temporary files problem. I think this has started since I have installed the latest version that I'm running, 1.11.8 (I'm running windows 10, 64 bits, FreePlane runs on Java 15.0.1.

I would be grateful if anybody has any ideas of how to correct this problem

And a happy new year to everybody in the group, but especially Dimitry and all the people actually programming FreePlane
Cheers
Oliver

**Comment by i-plasm** [comment]
I remember you asking about modifying creation date of nodes. My question is: did this problem start only after you tried scripting to change node creation dates, or is this an older problem?

**Comment by EdoFro** [comment]
Hi @Olifair,
Look at this preference: `Preferences…->Environment->Files->Experimental file locking`
I have it unchecked and have no problem.

Freeplane doesn't create temporary .mm~ for me. 
I think this option is only useful if the mindmaps are in a shared folder and the mindmaps can be opened by different users anytime.

**Comment by Olifair** [comment]
Hello, thanks for all the tips from @i-plasm @EdoFro and @dpolivaev ,
Dimitry seems to have accepted this as a bug, but just to answer some of the other questions (and for some reason I don't see a long comment by @i-plasm, which I got by email, anymore here on github) : 
* I run Freeplane directly by clicking on the app icon
* I do not have any particularly large maps, most are about 400kb, the biggest is 1Mb, but I don't have it open normally. 
* I have only one Freeplane instance open at any time
* this problem was already there before I started to play with the scripting

I followed @EdoFro suggestion to unselect the "Experimental file locking" and now I do not have the problem anymore. Is this an option that came with the last version ?

Thanks for all your help

Oliver

**Reply by i-plasm** [reply]
Thank you for answering. I deleted the message just to make sure it didn't have to do with modifying dates. Glad you fixed the problem, I had not heard about that option.
